backend/src/jobq_server/routers/builds.py

```diff
@@ -0,0 +1,153 @@
+import asyncio
+import io
+import shlex
+import tarfile
+import tempfile
+import uuid
+from copy import deepcopy
+from dataclasses import dataclass
+from enum import Enum
+from pathlib import Path
+from typing import Annotated
+
+from fastapi import APIRouter, BackgroundTasks, Depends, Form, HTTPException, UploadFile
+from fastapi.responses import JSONResponse, StreamingResponse
+from pydantic import BaseModel
+
+from jobq_server.utils.assembler import config
+from jobq_server.utils.assembler.renderers import RENDERERS
+from jobq_server.utils.processes import run_command
+
+router = APIRouter(tags=["Container image builds"])
+
+# In-memory storage for build jobs
+build_jobs: dict[str, dict] = {}
+
+
+class BuildJobStatus(str, Enum):  # noqa: UP042
```

Comment:
We could use `StrEnum` instead of `Str, Enum` server side, right?

Reply:
I don't see this code ending up in the real implementation, but in principle yes :)